### PR TITLE
Add retry logic to AzureStorageUtils.UploadBlockBlobAsync

### PR DESCRIPTION
**Pull request by MattGal**
## Change contents
Continuing to address https://github.com/dotnet/core-eng/issues/8582.  The verbosity of logging the Azure storage SDK team needs would massively increase our console log size so instead I'm simply adding retry around the API call.

## Some notes:

- As used here BlobClient.UploadAsync doesn't care if the blob already exists; so even if it's partially there (should not be the case; the call must finish to 'commit' the blob) the call will not fail.
- Azure Storage SDK confirms
  - They have no existing retry logic for network flakiness (only Azure throttling)
  - They have no retry policy built in like the <= V11 SDKs had so I used ours.
- The ExponentialRetry logic IS indeed in a weird namespace but I'm not moving it.

**TL;DR: Add 5 ExponentialRetry attempts per blob upload, and upgrade minor version too for luck.**

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs [hunk 1]
```diff
@@ -65,10 +66,41 @@ public async Task UploadBlockBlobAsync(string filePath, string blobPath)
...
+                    return true;
+                }
+                catch (System.Net.Http.HttpRequestException)
+                {
+                    // No logger hooked up: let other types of exceptions bubble up.
+                    // If we're here we have valid storage credentials and we're talking to Azure, 
+                    // so logging the exception is not super interesting (any non-HttpReq one still is)
+                    return false;
+                }
+            });
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs]: `.ConfigureAwait(false)` here too?

Comment (MattGal) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs]: Sorry I'm not following...  is there an async call that needs that tacked on the end that I'm missing?

Comment (MattGal) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs]: nm I get it. Though the usage I copy pasted didn't have it :)

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs [hunk 2]
```diff
@@ -65,10 +66,41 @@ public async Task UploadBlockBlobAsync(string filePath, string blobPath)
...
+                        filePath,
+                        headers)
+                        .ConfigureAwait(false);
+                    return true;
+                }
+                catch (System.Net.Http.HttpRequestException)
+                {
+                    // No logger hooked up: let other types of exceptions bubble up.
+                    // If we're here we have valid storage credentials and we're talking to Azure, 
+                    // so logging the exception is not super interesting (any non-HttpReq one still is)
```
[the remark this answers is not in the document]
Comment (MattGal) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs]: I honestly don't think that information is useful.  This is us calling an Azure SDK API where we already have validated that the storage account, its credentials, and the container to upload exist, throwing a specific platform API because they don't handle network flakiness.  I can try to store it off but honestly... it seems kind of useless to us or our users (sharing with the owning team I can certainly say that the exception object did not get them to fix anything)

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs (hunk 1 above)
Comment (dagood) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs]: Yeah, don't think I've done significant coding in a context where it matters before. Figured we should add it since we have it above though.

Comment (dagood) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs]: (Well, I guess the whole of VersionTools *should* have it since it could be used by anyone... but that was before I knew it was a thing.)

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (MattGal) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs]: OK it's now included as the inner exception when I throw the custom one letting the user know what happened.
